sharedcache/OSCacheConfigOptions.hpp

```diff
@@ -0,0 +1,121 @@
+/*******************************************************************************
+ * Copyright (c) 2020, 2020 IBM Corp. and others
+ *
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/
+ * or the Apache License, Version 2.0 which accompanies this distribution and
+ * is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * This Source Code may also be made available under the following
+ * Secondary Licenses when the conditions for such availability set
+ * forth in the Eclipse Public License, v. 2.0 are satisfied: GNU
+ * General Public License, version 2 with the GNU Classpath
+ * Exception [1] and GNU General Public License, version 2 with the
+ * OpenJDK Assembly Exception [2].
+ *
+ * [1] https://www.gnu.org/software/classpath/license.html
+ * [2] http://openjdk.java.net/legal/assembly-exception.html
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
+ *******************************************************************************/
+#if !defined(OS_CACHE_CONFIG_OPTIONS_HPP_INCLUDED)
+#define OS_CACHE_CONFIG_OPTIONS_HPP_INCLUDED
+
+#include "omr.h"
+
+class OSCacheConfigOptions
+{
+public:
+    typedef enum StartupReason {
+        StartupReason_Destroy,
+        StartupReason_Expired,
+        StartupReason_Open,
+        StartupReason_Stat,
+        StartupReason_EnsureWideOpen = 0x1000000
+    } StartupReason;
+
+    typedef enum StatReason {
+        StatReason_Destroy,
+        StatReason_Expired,
+        StatReason_GatherStatistics,
+        StatReason_Iterate,
+        StatReason_List,
+        StatReason_EnsureWideEnum = 0x1000000
+    } StatReason;
+
+protected:
+    I_32 _openMode;
+    StartupReason _startupReason;
+
+public:
+    OSCacheConfigOptions(I_32 openMode)
+        : _openMode(openMode)
+        , _startupReason(StartupReason::StartupReason_Open)
+    {}
+
+    virtual ~OSCacheConfigOptions() {}
+
+    virtual bool omr_useUserHomeDirectoryForCacheDir() = 0;
+    virtual bool omr_isUserSpecifiedCacheDir() = 0;
+    virtual bool omr_readOnlyOpenMode() = 0;
+
+    virtual I_32 omr_fileMode();
+    virtual I_32 omr_openMode() = 0;
+    virtual UDATA omr_groupPermissions() = 0;
+
+    virtual IDATA omr_cacheDirPermissions() = 0;
+    virtual bool omr_usingNetworkCache() = 0;
+
+    /* TODO: the restore check only applies to the shared memory cache,
```

Comment:
The format of this comment (and multi-line comments in `OSCacheRegion.hpp`) are not correct, see the coding standard for comment: 
https://github.com/eclipse/omr/blob/master/doc/CodingStandard.md#formatting-comments

Reply:
I have updated this to a doxygen comment in the latest commit. Thanks :)